src/hooks/useCanUseTouchScreen.js

```diff
@@ -0,0 +1,19 @@
+import {useEffect, useState} from 'react';
+import * as DeviceCapabilities from '../libs/DeviceCapabilities';
+
+/**
+ * This hooks returns a boolean of whether the device is a touch screen or not.
+ * @returns {Boolean}
+ */
+export default function useCanUseTouchScreen() {
+    const [canUseTouchScreen, setCanUseTouchScreen] = useState(DeviceCapabilities.canUseTouchScreen());
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    useEffect(() => {
+        const newCanUseTouchScreen = DeviceCapabilities.canUseTouchScreen();
+        if (canUseTouchScreen === newCanUseTouchScreen) return;
+        setCanUseTouchScreen();
+    });
+
+    return canUseTouchScreen;
```

Comment:
You can achieve the same outcome with a more straightforward approach:
```suggestion
    return DeviceCapabilities.canUseTouchScreen();
```

Here's why this alternative may be beneficial:
1. The current implementation uses a `useEffect` that runs logic on each render (since there's no dependencies array).
2. In such a scenario, you could simply execute the logic in the hook body, eliminating the need for a `useEffect` and `state`
```